### PR TITLE
fix(crux): fix secret disappearance

### PR DESCRIPTION
**Pull request by balanceee**

### CODE REVIEW

**Review by robot9706 — APPROVED** [review]
LGTM 😎

**Review by polaroi8d — APPROVED** [review]

**Review by polaroi8d — APPROVED** [review]

File: web/crux-ui/src/models/container.ts [hunk 1]
```diff
@@ -341,32 +336,30 @@ export const mergeConfigs = (
...
-
   return {
-    // common
-    name: instanceConfig.name || imageConfig.name,
-    environment: envs,
-    secrets: instanceConfig?.secrets ? instanceConfig.secrets : imageConfig.secrets,
-    ports: overridePorts(imageConfig?.ports, instanceConfig.ports),
+    // Default: override
+    // UniqueKeyValues/KeyValues: merge
+    name: override(imageConfig?.name, instanceConfig.name),
```
Comment (m8vago) [inline web/crux-ui/src/models/container.ts:342]: isn't like the instanceConfig could be null and not the other way around?

**Review by robot9706 — APPROVED** [review]
LGTM 📚

File: web/crux-ui/src/models/container.ts [hunk 2]
```diff
@@ -341,32 +336,30 @@ export const mergeConfigs = (
...
+    environment: mergeKeyValues(imageConfig.environment, instanceConfig.environment),
```
Comment (m8vago) [inline web/crux-ui/src/models/container.ts:343]: `instanceConfig?.environment`

File: web/crux/src/app/deploy/deploy.mapper.ts [hunk 3]
```diff
@@ -325,69 +324,81 @@ export default class DeployMapper {
...
       proxyHeaders: this.override(imageConfig?.proxyHeaders, instanceConfig?.proxyHeaders),
       extraLBAnnotations: this.override(imageConfig?.extraLBAnnotations, instanceConfig?.extraLBAnnotations),
       healthCheckConfig: this.override(imageConfig?.healthCheckConfig, instanceConfig?.healthCheckConfig),
       resourceConfig: this.override(imageConfig?.resourceConfig, instanceConfig?.resourceConfig),
       useLoadBalancer: this.override(imageConfig?.useLoadBalancer, instanceConfig?.useLoadBalancer),
       deploymentStrategy: this.override(imageConfig?.deploymentStrategy, instanceConfig?.deploymentStrategy),
-      labels: this.override(imageConfig?.labels, instanceConfig?.labels),
-      annotations: this.override(imageConfig?.annotations, instanceConfig?.annotations),
+      labels:
+        imageConfig?.labels || instanceConfig?.labels
```
Comment (m8vago) [inline web/crux/src/app/deploy/deploy.mapper.ts:374]: instance should be stronger than the image, i think

**Review by m8vago — APPROVED** [review]

**Comment by polaroi8d** [comment]
Tested in local, works. 🤔

**Pull request merged**